k9mail/src/main/java/com/fsck/k9/planck/ui/fragments/AccountSetupBasicsFragmentBase.kt

```diff
@@ -61,6 +65,23 @@ abstract class AccountSetupBasicsFragmentBase : Fragment() {
         restoreDataState(savedInstanceState)
     }
 
+    protected fun setupToolbar(showBackArrow: Boolean = true) {
+        (requireActivity() as AccountSetupBasics).initializeToolbar(
+            !requireActivity().isTaskRoot
+                    || parentFragmentManager.backStackEntryCount > 1,
+            R.string.account_setup_basics_title
+        )
+    }
+
+    private fun handleSignInResult(resultCode: Int) {
+        if (resultCode == Activity.RESULT_CANCELED) {
+            deleteAccount()
```

Comment:
in my understanding, if the user cancels the sign-in, we delete the account being added. Is it potentially dangerous in case we call this function later from a re-auth flow? I know in there we don't call the AccountSetupBasicsFragment, but maybe we will forget and use this method and it could lead to problems. I don't know if there is a check we can make to ensure delete is only called if account is new

Reply:
Ah I see your point.
Yes, at this point int these screens the account can only be new. Which means it is not fully setup, and we can just delete it from Preferences. For an account that has been already saved etc more work is needed, so since we know there will be no issues we just delete the account in the making and it will be re-created with fresh settings when it is needed. In this way we also ensure we dont get any inconsistent situation with mail settings for the account.